File: 134_GasStation/python/GasStation.py
class Solution:
    def canCompleteCircuit(self, gas: List[int], cost: List[int]) -> int:
        if len(gas) == 0:
            return -1

        
        dif = [gas[i]-cost[i] for i in range(len(gas))]
        
        if sum(dif)<0:
            return -1
        
        
        rest = 0
        ans = 0
        for i in range(len(gas)):
            rest+=dif[i]
            if rest<0:
                rest = 0
                ans = i+1
                    
        return ans

File: 134_GasStation/c++/GasStation.cc
class Solution {
public:
    int canCompleteCircuit(vector<int>& gas, vector<int>& cost) {
        if(gas.size()==0){
            return -1;
        }
        
        int ans = 0;
        int totalDif = 0;
        int restGas = 0;
        
        for(int i = 0; i<gas.size(); i++){
            totalDif+=gas[i]-cost[i];
            restGas+=gas[i]-cost[i];
            if(restGas<0){
                restGas = 0;
                ans = i+1;
            }
        }
        return totalDif<0?-1:ans;
    }
};